paths_cli/wizard/wizard.py

```diff
@@ -313,13 +313,33 @@ def run_wizard(self):
         # TODO: next line is only temporary
         self.say("Today I'll help you set up a 2-state TPS simulation.")
         self._patch()  # try to hide the slowness of our first import
-        for step in self.steps:
-            req = step.store_name, step.minimum, step.maximum
-            do_another = True
-            while do_another:
-                do_another = self._do_one(step, req)
-        storage = self.get_storage()
-        self.save_to_file(storage)
+        try:
+            for step in self.steps:
+                req = step.store_name, step.minimum, step.maximum
+                do_another = True
+                while do_another:
+                    do_another = self._do_one(step, req)
+        except QuitWizard:
+            do_save = self._ask_save()
+        else:
+            do_save = True
+
+        if do_save:
+            storage = self.get_storage()
+            self.save_to_file(storage)
+        else:
+            self.say("Goodbye! 👋")
```

Comment:
Don't we say goodbye if we safe files? If so you might want to consider:
```suggestion
        if do_save:
            storage = self.get_storage()
            self.save_to_file(storage)
        self.say("Goodbye! 👋")
```

Reply:
I considered this when adding that, but the main reason for the "goodbye" was ending when not saving felt abrupt. (Wizard says, "Do you want save progress", you say no, then you're dumped into your shell -- saying "goodbye" makes the exit clearer.) If you save to a file, the last thing it does is to tell you the contents of the file, so that gives a reasonable conclusion to the interaction.

I'll leave as-is for now, but I'm indifferent to this -- if later someone feels that the "goodbye" is also better after saving, then I'm open to the change.